### PR TITLE
Added parameters to scheduler

### PR DESCRIPTION
**Pull request by bhasin85**
# WIP

### CODE REVIEW

**Comment by akashisama** [comment]
We'll need to make sure the scheduling times are alterable; we can't have both systest and prod seeders hammering the database at the same time.

Any ideas on what's the best way to get the cloudwatch JSON values into the main function as parameters? I mean we could just call the function with the JSON values, but it'd be nice to somehow keep the defaults that are defined in the click interface.

**Comment by akashisama** [comment]
@lwgordonimos if you look at my last commit on this, I changed the lambda handler to call a plain-old python function rather than the click-enhanced function. The problem was that when calling the click function from code we were getting errors about the number of parameters. Does this seem like the best solution to you or am is there something more pythony?

**Comment by lwgordonimos** [comment]
As discussed, the only real problem I can see is that since click does type validation and also type casting, we need to make sure invalid types get rejected at the entry point and not halfway into the code when something tries to use the variable.

If we can assume CloudFormation has already done the validation, we can just add these kinds of statements as a last check and let it throw to Lambda unhandled, since it *should* never happen.

```python
trace_count = int(event['trace_count']) # for INT params
assert event['req_type'] in [TRUNCATE, PURGE, SEED], "invalid request type" # for Choice params
```

**Comment by akashisama** [comment]
@lwgordonimos changes made

File: cloudformation/lambda/geowebcache_seeder/geowebcache_seeder.py [hunk 1]
```diff
@@ -71,8 +78,13 @@ def get_final_layers(geoserver, layers, wms_layers):
 
 # AWS Lambda Handler
 def handler(event, context):
+    assert event['req_type'] in [TRUNCATE, SEED, PURGE], "Request type {type} not recognised".format(type=event['req_type'])
+    start_zoom = int(event['start_zoom'])
+    end_zoom = int(event['end_zoom'])
+    thread_count = event['thread_count']
```
Comment (lwgordonimos) [inline cloudformation/lambda/geowebcache_seeder/geowebcache_seeder.py]: int() missing?

**Comment by akashisama** [comment]
Gah, good catch

**Pull request merged**